src/components/DocumentRenderer/DocumentDetailsSheet.tsx

```diff
@@ -13,16 +15,41 @@ export const DocumentDetailsSheet: FunctionComponent<DocumentDetailsSheet> = ({
   document
 }) => {
   const [headerHeight, setHeaderHeight] = useState(0);
+  const hasHeaderHeightBeenSet = useRef(false);
   const onHeaderLayout = (event: LayoutChangeEvent): void => {
-    const { height } = event.nativeEvent.layout;
-    setHeaderHeight(height + 56);
+    if (!hasHeaderHeightBeenSet.current) {
+      const { height } = event.nativeEvent.layout;
+      setHeaderHeight(height + 56);
+      hasHeaderHeightBeenSet.current = true;
+    }
   };
 
   const { name } = document.data.issuers[0];
   const issuerName = name.split(":")[2];
 
+  const [tamperedCheck, setTamperedCheck] = useState<CheckStatus>("checking");
+  const [issuedCheck, setIssuedCheck] = useState<CheckStatus>("checking");
+  const [revokedCheck, setRevokedCheck] = useState<CheckStatus>("checking");
+  const [issuerCheck, setIssuerCheck] = useState<CheckStatus>("checking");
+
+  useEffect(() => {
```

Comment:
Possible to move this into the banner so that the interface with the banner is simply 
`<ValidityBanner document={document} />`

Reason is that I'll need this in the other action sheet as well when scanning a document, will want to reuse this

Reply:
I've moved the checking logic to a React hook: `useDocumentVerifier` instead. So the verifier logic isn't coupled with `ValidityBanner`, which implements one approach in displaying the validity.